R/bundlePackage.R

```diff
@@ -0,0 +1,157 @@
+    name <- deps[i, "Package"]
+
+    # get package info
+    info <- as.list(deps[i, c("Source", "Repository")])
+
+    # include github package info
+    info <- c(info, as.list(deps[i, grep("Github", colnames(deps), perl = TRUE, value = TRUE)]))
+
+    # get package description; note that we need to remove the
+    # packageDescription S3 class from the object or jsonlite will refuse to
+    # serialize it when building the manifest JSON
+    # TODO: should we get description from packrat/desc folder?
+    info$description <- suppressWarnings(unclass(utils::packageDescription(name)))
+
+    # TODO(HW): should this code be removed? I don't see how it is ever reached
+    # if description is NA, application dependency may not be installed
+    if (is.na(info$description[1])) {
+      errorMessages <- c(
+        errorMessages,
+        paste0(
+          capitalize(assetTypeName), " depends on package \"", name, "\"",
+          "but it is not installed. Please resolve before continuing."
+        )
+      )
+      next
+    }
+
+    # validate package source (returns a message if there is a problem)
+    packageMessages <- c(packageMessages, validatePackageSource(deps[i, ]))
+
+    # good to go
+    packages[[name]] <- info
+  }
+
+  if (length(packageMessages)) {
+    # Advice to help resolve installed packages that are not available using the
+    # current set of configured repositories. Each package with a missing
+    # repository has already been printed (see snapshotDependencizes).
+    #
+    # This situation used to trigger an error (halting deployment), but was
+    # softened because:
+    #   * CRAN-archived packages are not visible to our available.packages
+    #     scanning.
+    #   * Source-installed packages may be available after a manual server-side
+    #     installation.
+    #
+    # That said, an incorrectly configured "repos" option is almost always the
+    # cause.
+    packageMessages <- c(
+      packageMessages,
+      paste0(
+        "Unable to determine the source location for some packages. ",
+        "Packages should be installed from a package repository like ",
+        "CRAN or a version control system. Check that ",
+        "options('repos') refers to a package repository containing ",
+        "the needed package versions."
+      )
+    )
+    warning(paste(formatUL(packageMessages, "\n*"), collapse = "\n"), call. = FALSE, immediate. = TRUE)
+  }
+
+  if (length(errorMessages)) {
+    stop(paste(formatUL(errorMessages, "\n*"), collapse = "\n"), call. = FALSE)
+  }
+
+  packages
+}
+
+## check for extra dependencies congruent to application mode
+inferRPackageDependencies <- function(appMode,
+                                      hasParameters,
+                                      documentsHavePython,
+                                      quartoInfo) {
+
+  # TODO(HW): appUsesR(quartoInfo) is always TRUE because this function is
+  # only called for apps that use R
```

Comment:
I think it's perhaps still better to leave that call on line 120 than to remove it (assuming I understand what you're talking about). Although it will always be `TRUE` where the function is called within `bundlePackages`, I'd rather have that unnecessary call than have this function return incorrect R dependencies for Quarto content that doesn't use R.

Reply:
My feeling was that the name of the function already gives a pretty strong signal — why would you be calling `inferRPackageDependencies` for content that doesn't use R? But if you want this to be robust against future incorrect usage, I could error if `!appUsesR.

Reply:
The call to this function is now guarded by `appUsesR`; I'm OK making that an expectation, and leave it to the caller to understand what information needs to be known before calling. By combining the call to `inferredRDependencies` and `shapshotRDependencies`, we need to look at `appUsesR` much less.